sdk/core/System.Net.ClientModel/tests/client/MapsClient/MapsClient.cs

```diff
@@ -82,7 +79,12 @@ public virtual Result GetCountryCode(string ipAddress, RequestOptions options =
 
     private ClientMessage CreateGetLocationRequest(string ipAddress, RequestOptions options)
     {
-        ClientMessage message = _pipeline.CreateMessage();
+        MessagePipeline pipeline = options.GetPipeline();
+        ClientMessage message = pipeline.CreateMessage();
+
+        // TODO: this overrides anything the caller passed, so we need to fix that.
+        // Note: this throws right now because options are frozen.
+        options.MessageClassifier = new ResponseStatusClassifier(stackalloc ushort[] { 200 });
```

Comment:
We'll want to either make this non-freezable on options, or find another way to set per-operation message classifiers.  If we make it non-freezable, that makes "freezing options" harder to understand/explain, because only some options are frozen, and some are not.

Precedence rules for MessageClassifier are a bit complex, e.g. they can be set these ways:
- Client constructor via PipelineOptions
  - passed-in by client-user
  - added by client-author in constructor
  - constructed default if not specified another way
- Protocol method via RequestOptions
  - passed-in by client-user
  - added by client-author in method
  - constructed default if not specified another way
- CreateRequest private method
  - added by generator to set operation-specific success status codes
- Options.Apply method
  - copied to message from options
- On ClientMessage via public MessageClassifier property
  - This could be set by a client-user or client-author in a custom policy

Reply:
We're moving in the right direction on this - so far, we've removed the following ways to set MessageClassifier:
- On PipelineOptions
- On the ClientMessage

We still need to solve how we will prevent client authors from accidentally blowing away client-user passed MessageClassifiers.  I believe that for precedence rules, anything an end user passes should override generated defaults, but we may want to introduce the classifier chaining handlers we have in Azure.Core so that end users need to explicitly override status codes and have those sets merged in the client before options are applied to the message.

Reply:
I added precedence rules such that the MessageClassifier added to the Message is the first specified of "either the client-users's MessageClassifier or the client-author's MessageClassifier or the default MessageClassifier".  We still need to decide if we want ClientModel clients to use classifier chaining the same way we do in Azure.Core-based clients.